examples/package_build/build/build.odin

```diff
@@ -0,0 +1,75 @@
+package src_build
+
+import "core:build"
+import "core:os"
+import "core:strings"
+
+Mode :: enum {
+    Debug,
+    Release,
+}
+
+Target :: struct {
+    using target: build.Target,
+    mode: Mode,
+}
+
+project: build.Project
+target_debug: Target = {
+    target = {
+        name = "deb",
+        platform = {ODIN_OS, ODIN_ARCH},
+    },
+    mode = .Debug,
+}
+target_release: Target = {
+    target = {
+        name = "rel",
+        platform = {ODIN_OS, ODIN_ARCH},
+    },
+    mode = .Release,
+}
+
+config_target :: proc(project: ^build.Project, target: ^build.Target) -> (config: build.Config) {
+    target := cast(^Target)target
+    config.name = target.name 
+    config.platform = target.platform
+    config.out_file = "demo.exe" if target.platform.os == .Windows else "demo.out"
+    config.out_dir = strings.concatenate({"out/", target.name})
+    config.build_mode = .EXE
+    config.src_path = "src"
```

Comment:
As far as I'm aware this is a required option. Maybe print error when this isn't specified?

Reply:
Error handling is still dumb. Will start making the code more friendly in that regard